### ISSUE TITLE
false positive @typescript-eslint/no-unsafe-assignment when deriving prop type from ComponentProps

### ISSUE THREAD

**Issue by Theo-Steiner** [sveltejs/eslint-plugin-svelte#1303]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.34.0

### What version of `eslint-plugin-svelte` are you using?

3.11.0

### What did you do?

<details>
<summary>Configuration</summary>

```
import js from "@eslint/js";
import sveltePlugin from "eslint-plugin-svelte";
import tseslint from "typescript-eslint";
export default tseslint.config(
  js.configs.recommended,
  ...tseslint.configs.recommended,
  ...sveltePlugin.configs["flat/recommended"],
  {
    files: ['**/*.svelte', '**/*.svelte.ts', '**/*.svelte.js'],
    languageOptions: {
      parserOptions: {
        parser: tseslint.parser,
        extraFileExtensions: [".svelte"],
        projectService: true
      },
    }
  },
  {
  	rules: {
			'@typescript-eslint/no-unsafe-assignment': 'error',
		}
  }
);
```
</details>

ComponentA.svelte
```svelte
<script lang="ts">
    let {a}: {a: number} = $props();
</script>

<p>{a}</p>
```

MyComponent.svelte ← error occurs here
```svelte
<script lang="ts">
    import type { ComponentProps } from 'svelte';

    import ComponentA from './ComponentA.svelte'

    interface Props {
        a: ComponentProps<typeof ComponentA>['a']
    }

    let { a }: Props = $props()
</script>

<ComponentA {a} />
```

### What did you expect to happen?

`a` should be inferred correctly as `number` based on the typings of `ComponentA`

### What actually happened?

The other component's prop instead is inferred as any: `Unsafe object destructuring of a property with an `any` value.`

### Link to **GitHub Repo** with Minimal Reproducible Example

[Playground](https://eslint-online-playground.netlify.app/#eNqNU01v2zAM/SuCMEBtENt3Nw067OM2rMCOUQ+aTQf2bEmQ5CBF4P8+ypIdJ+m6OpeIenzke6RO1Joi+/H6RXVaSZAutQdoHdCcbmxhau1IK+T+kVNnOd1ySfCrEWscca8ayInMqc9GaUsGUhnVERZ42AOXF0kz+nPEpdk5FIuzOUc6MJUogATuUwj7T+RXhTe+HVUtCmx3TLCXkDJMlC047FmQIY+cj+ST9n/u7rncZEEz6sTDotWTGEi2pWt0oVCyqvdpY5VEk8aOOK1l0fYlcJqTXajDR2NXURGn6zk8BtIy9Yb6IHY4ILOH3zjx7hhGKQKFnNAM2Xe/wQwLOej8pR69RfQm014H2BbNTWcxWCgOqLFhMJw+BVDWYEkki/ehsee239dyQgZgosdgMkk+5zgbEBPejyp0loSLEQzHEVxCJfr2nBSbvPOiGxtPNjVQqK4DWUI5epum6OhFxpuYZfsTbsdp1QqXLeCcvowZceOqugWLs2Wr1TxTtibLI1a/jjSWBRacFU6vF3v4qV2tJFLNq6yFsWBu49NNfjYiBCKj/+DojPiOvX07OpA2MKCaeeum8iOdUQ0U7heYQ10A0poeptsh4vCZTIew2QgaleOJO/9jTzfDy6RKemlFBYmwtt5LtNCxnDAwRhmGbD4xUHN5/4Drp0XxB924ekWe2T8hTjtVYt34ajgt4fAVtB+MLGrAdZx9utxSn4qDBIsLNT+4acPeubze3DehH7iMhiRhUP+Anv37L2B+HUtc9HGgw1+hiO4t)
